### ISSUE TITLE
The epoch argument is wrong

### ISSUE THREAD

**Issue by wlnirvana** [spro/char-rnn.pytorch#6]
The `--n_epochs` argument is in fact an iteration, which updates the weight based on a random sample of data points.

There is no conclusive definition, but the community generally define epoch to be an entire pass of the whole dataset. See [this](https://deeplearning4j.org/glossary#epoch-vs-iteration) and [this](https://www.quora.com/What-is-an-epoch-in-deep-learning)

**Comment by samlanka** [comment]
Yes, the code is counting the number of iterations, not number of epochs
